tfexec/terraform.go

```diff
@@ -203,11 +212,21 @@ terraform {
 		return nil, fmt.Errorf("failed to create override file: %s", err)
 	}
 
+	// create local workspace state directory
+	workspaceStatePath := filepath.Join(c.Dir(), "terraform.tfstate.d", workspace)
+	workspacePath := filepath.Join(c.Dir(), "terraform.tfstate.d")
+	log.Printf("[INFO] [migrator@%s] creating local workspace folder in: %s\n", c.Dir(), workspaceStatePath)
+	if err := os.MkdirAll(workspaceStatePath, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("failed to create local workspace state directory: %s", err)
+	}
+
 	log.Printf("[INFO] [executor@%s] switch backend to local\n", c.Dir())
 	err := c.Init(ctx, "", "-input=false", "-no-color", "-reconfigure")
 	if err != nil {
 		// remove the override file before return an error.
 		os.Remove(path)
+		os.Remove(workspacePath)
+		os.Remove(workspaceStatePath)
```

Comment:
Should we remove the child `workspaceStatePath` before removing the parent `workspacePath`?

Reply:
yes, thank you.